File: java/program52.java
import java.util.*;

class NumberX
{
    public int No;

    NumberX(int no)
    {
        this.No = no;
    }

    public int ReverseNumber()
    {
        int iDigit = 0;
        int result = 0;
        int temp = No;

        while(temp != 0)
        {
            iDigit = temp % 10;
            result = (result*10) + iDigit;
            temp = temp / 10;
        }
        return result;
    }
}

class program52
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number");
        int value = sobj.nextInt();

        NumberX obj = new NumberX(value);
        int iRet = obj.ReverseNumber();

        System.out.println("Reverse number is : "+iRet);
    }   
}

File: cpp/program52.cpp
// program63.c

#include<iostream>
using namespace std;

class Digit 
{
    public:
        int iNo;

        Digit(int X)
        {
            iNo = X;
        }

        int ReverseNumber()
        {
            int iDigit = 0;
            int iResult = 0;
            int temp = 0;

            if(iNo < 0)
            {
                iNo = -iNo;
            }

            temp = iNo;
            while(temp != 0)
            {
                iDigit = temp % 10;
                iResult = (iResult * 10) + iDigit;
                temp = temp / 10;
            }
            return iResult;
        }
};

int main()
{
    int iValue = 0;
    int iRet = 0;

    cout<<"Enter number \n";
    cin>>iValue;

    Digit obj(iValue);

    iRet = obj.ReverseNumber();
    cout<<"Reverse number is : "<<iRet<<"\n";

    iRet = obj.ReverseNumber();
    cout << "Reverse number is : " << iRet << "\n";

    return 0;
}
